src/main/java/com/minecolonies/blocks/AbstractBlockHut.java

```diff
@@ -83,12 +82,31 @@ public TileEntity createNewTileEntity(World world, int meta)
         return new TileEntityColonyBuilding();
     }
 
+    /**
+     * Convert the given metadata into a BlockState for this Block
+     */
+    @NotNull
     @Override
-    public int getMetaFromState(@NotNull IBlockState state)
+    public IBlockState getStateFromMeta(int meta)
     {
-        return state.getValue(FACING).getIndex();
+        EnumFacing enumfacing = EnumFacing.getFront(meta);
+
+        if (enumfacing.getAxis() == EnumFacing.Axis.Y)
+        {
+            enumfacing = EnumFacing.NORTH;
+        }
+
+        return this.getDefaultState().withProperty(FACING, enumfacing);
     }
 
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)